phd-tests/testcase_macro/src/lib.rs

```diff
@@ -27,23 +29,29 @@ pub fn phd_testcase(_attrib: TokenStream, input: TokenStream) -> TokenStream {
             phd_testcase::TestCase::new(
                 module_path!(),
                 #fn_name,
-                phd_testcase::TestFunction { f: #fn_ident }
+                phd_testcase::TestFunction { f: |ctx| Box::pin(#fn_ident(ctx)) }
             )
         }
     };
 
+    if item_fn.sig.asyncness.is_none() {
+        abort!(item_fn.sig.span(), "PHD test cases must be async");
+    }
+
     // Rebuild the test body into an immediately-executed function that returns
     // an `anyhow::Result`. This allows tests to use the `?` operator and to
     // `return Ok(())` to allow a test to pass early.
     let fn_vis = item_fn.vis.clone();
     let fn_sig = item_fn.sig.clone();
-    let fn_block = item_fn.block;
+    let fn_block = item_fn.block.stmts;
     let remade_fn = quote! {
+        #[tracing::instrument(level = "info", name = #fn_name, skip_all)]
```

Comment:
I'm playing around with this and I think it might also make sense to include the module path in the span name, since it's the combination of module path and function name that produce a fully-qualified test case. (Without this there are liable to be multiple spans named `smoke_test` or something like that--okay, but in which module?)

Reply:
Yeah, that's a good call. You could do 
```suggestion
        #[tracing::instrument(level = "info", name = concat!(module_path!(), #fn_name), skip_all)]
```
which...at least, I _think_ should work?

Reply:
> This doesn't quite build; I _think_ the issue is that `tracing_attribute::InstrumentArgs` [requires](https://github.com/tokio-rs/tracing/blob/baeba47cdaac9ed32d5ef3f6f1d7b0cc71ffdbdf/tracing-attributes/src/attr.rs#L18-L30) `name` to be a `syn::LitStr`, which an invocation of `concat!` is not (it's a `syn::Macro` or suchlike).

hahaha...shows what I know about my own library! :P

another option, which might be a bit prettier, is: 
```rust
#[tracing::instrument(level = "info", name = "test", fields(message = %concat!(module_path!(), "::", #fn_name), skip_all)]
```

which would format like: `test{phd_tests::smoke::nproc_test}`

Reply:
Touched up in 04b48c37.